### PR TITLE
Split Android specific code out of PlatformEGL

### PR DESCRIPTION
**Pull request by pixelflinger**
We now should be able to use PlatformEGL for all generic EGL platforms
(not tested though). Most texture streaming operations are not supported
in that case, since they're platform specific.

### CODE REVIEW

File: filament/backend/src/opengl/PlatformEGLAndroid.cpp [hunk 1]
```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (C) 2017 The Android Open Source Project
```
Comment (prideout) [inline filament/backend/src/opengl/PlatformEGLAndroid.cpp]: We should probably update the year when creating a new file.

File: filament/backend/src/opengl/PlatformEGL.cpp [hunk 2]
```diff
@@ -9,44 +9,28 @@
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * WITHOUT WARANTIES OR CONDITIONS OF ANY KIND, either express or implied.
```
Comment (bejado) [inline filament/backend/src/opengl/PlatformEGL.cpp]: `WARRANTIES` is correct

Comment (pixelflinger) [reply filament/backend/src/opengl/PlatformEGL.cpp]: oh that's weird. I don't know how it got changed. good catch.

Comment (pixelflinger) [inline filament/backend/src/opengl/PlatformEGL.cpp]:
```suggestion
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
```

File: filament/backend/src/opengl/PlatformEGLAndroid.cpp (hunk 1 above)
Comment (pixelflinger) [inline filament/backend/src/opengl/PlatformEGLAndroid.cpp]:
```suggestion
 * Copyright (C) 2019 The Android Open Source Project
```

**Pull request merged**